apps/sms/desktop-mock/navigator_moz_mobilemessage.js

```diff
@@ -1538,9 +1537,7 @@
   };
 
   MockNavigatormozMobileMessage.retrieveMMS = function(id) {
-    var request = {
-      error: null
-    };
+    var deferred = Utils.Promise.defer();
```

Comment:
__nit:__ shim-host.html includes this file, but doesn't include ```utils.js``` - so maybe we can just use ```new Promise``` here (e.g. instead of ```setTimeout``` below since we don't need timeout now anyway).